src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2209,7 +2209,164 @@ void *instanceservicesession(void *dummyPt) {
+            for (int i = 0; i < orgCount; i++) {
+                workerThreads[count++] = std::thread(JasmineGraphInstanceService::collectBucketstoCoordinator,
+                                                     sessionargs, orgHostList[i], stoi(orgPortList[i]),
+                                                     stoi(orgPortList[i]), graphID, noClusters);
+                sleep(1);
+            }
+
+            for (int threadCount = 0; threadCount < count; threadCount++) {
+                if(workerThreads[threadCount].joinable()) {
+                    workerThreads[threadCount].join();
+                }
+                std::cout << "Thread [A]: " << threadCount << " joined" << std::endl;
+            }
+        }
+        else if (line.compare(JasmineGraphInstanceProtocol::SHARE_BUCKETS_TO_COORDINATOR) == 0) {
+            instance_logger.log("Received : " + JasmineGraphInstanceProtocol::SHARE_BUCKETS_TO_COORDINATOR, "info");
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+
+            //Read coordinator host name
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string coordinatorHost = (data);
+            coordinatorHost = utils.trim_copy(coordinatorHost, " \f\n\r\t\v");
+            instance_logger.log("Received HostName: " + coordinatorHost, "info");
+
+            //Read coordinator port from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string coordinatorPort = (data);
+            coordinatorPort = utils.trim_copy(coordinatorPort, " \f\n\r\t\v");
+            instance_logger.log("Received Port: " + coordinatorPort, "info");
+
+            //Read coordinator data port from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string coordinatorDataPort = (data);
+            coordinatorDataPort = utils.trim_copy(coordinatorDataPort, " \f\n\r\t\v");
+            instance_logger.log("Received Data Port: " + coordinatorDataPort, "info");
+
+            //Read graph ID from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string graphID = (data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
+
+            //Read cluster count from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            int noClusters = stoi(data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
+
+            for (int i = 0; i < noClusters; i++) {
```

Comment:
This `noClusters` should be number of buckets?

Reply:
The count should be the worker count, variable name changed to reflect that

Reply:
Ok